legend-depot-core-services/src/main/java/org/finos/legend/depot/services/projects/ProjectsServiceImpl.java

```diff
@@ -118,7 +130,13 @@ public void checkExists(String groupId, String artifactId) throws IllegalArgumen
     @Override
     public void checkExists(String groupId, String artifactId, String versionId) throws IllegalArgumentException
     {
-        if (!this.projectsVersions.find(groupId,artifactId,versionId).isPresent())
+        Optional<StoreProjectVersionData> excludedProjectVersion = this.projectsVersions.findExcludedVersion(groupId, artifactId, versionId);
```

Comment:
this code can be made so much simple... simply get the project version... if is excluded then throw the error otherise flow is as  bfore...

u dont need that api at all